frontends/web/src/components/badge/badge.tsx

```diff
@@ -14,22 +14,33 @@
  * limitations under the License.
  */
 
-import { ReactNode } from 'react';
+import { ReactElement, ReactNode } from 'react';
 import style from './badge.module.css';
 
 type TBadgeStyles = 'success' | 'warning'; // TODO: not yet implemented 'info' | 'danger'
 
 type TProps = {
-  children: ReactNode;
+  children?: ReactNode;
   type?: TBadgeStyles;
-}
+  variation?: 'icon-only';
```

Comment:
Instead of being a prop, do you think it makes sense if we derive this from the `children` prop instead?

E.g, if children is undef, then it's icon-only. Otherwise, we shouldnt apply icon-only class.

Bc applying `variation=icon-only` w/ `children` is never be the case if i'm not mistaken.

Reply:
ack, will change this.